tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -246,29 +240,26 @@ LogicalResult ConvertTFLRelu6Op::matchAndRewrite(
         output_type.getElementType()
             .dyn_cast<mlir::quant::UniformQuantizedType>();
 
-    int64_t rescaled_6 = std::llround(6.0f / output_qtype.getScale());
+    int64_t rescaled_6 = std::llround(6.0f / output_qtype.getScale()) +
+                         output_qtype.getZeroPoint();
 
     Value op1_rescale_in =
-        buildRescaleToInt32(rewriter, op, tfl_relu6_op.x(),
-                            input_qtype.getScale() / output_qtype.getScale(),
-                            input_qtype.getZeroPoint());
-    auto op2_relun_op1 = rewriter.create<tosa::ReluNOp>(
-        op->getLoc(), rescale_type, op1_rescale_in,
-        rewriter.getI64IntegerAttr(rescaled_6), rewriter.getF32FloatAttr(0.0f));
-    Value op3_rescale_op2 = buildRescaleFromInt32(
-        rewriter, op, output_type, op2_relun_op1.getResult(), 1.0f,
-        output_qtype.getZeroPoint());
-
-    output = op3_rescale_op2;
+        buildRescale(rewriter, op, output_type, tfl_relu6_op.x(),
+                     input_qtype.getScale() / output_qtype.getScale(),
+                     input_qtype.getZeroPoint(), output_qtype.getZeroPoint(),
+                     /*double_round=*/false, /*scale32=*/true);
+    rewriter.replaceOpWithNewOp<tosa::ClampOp>(
+        op, output_type, op1_rescale_in,
+        rewriter.getI64IntegerAttr(output_qtype.getZeroPoint()),
+        rewriter.getI64IntegerAttr(rescaled_6), rewriter.getF32FloatAttr(0.0f),
+        rewriter.getF32FloatAttr(0.0f));
   } else {
-    auto op1_relun_in = rewriter.create<tosa::ReluNOp>(
-        op->getLoc(), output_type, tfl_relu6_op.x(),
-        rewriter.getI64IntegerAttr(0), rewriter.getF32FloatAttr(6.0f));
-
-    output = op1_relun_in.getResult();
+    rewriter.replaceOpWithNewOp<tosa::ClampOp>(
```

Comment:
I think you can refactor this to include only a single clamp by computing the integer min/max and adding a rescale in a conditional. It would allow us to include only a sing construction of the tosa::ClampOp rather than branching on the full behavior.

You just need to set the min/max at 0 and 6, then compute other values if its min/max.

Reply:
Done.